pymatgen/electronic_structure/dos.py

```diff
@@ -1165,6 +1166,166 @@ def get_upper_band_edge(
         upper_band_edge = energies[np.argmax(densities)]
         return upper_band_edge
 
+    def get_dos_fp(
+        self,
+        type: str = "summed_pdos",
+        binning: bool = True,
+        min_e: float | None = None,
+        max_e: float | None = None,
+        n_bins: int = 256,
+        normalize: bool = True,
+    ):
```

Comment:
Missing return type.

Reply:
I have added the type annotation (namedtuple) here now.